lib/saml_idp/saml_response.rb

```diff
@@ -18,21 +18,26 @@ class SamlResponse
     attr_accessor :encryption_opts
     attr_accessor :session_expiry
     attr_accessor :signed_message_opts
+    attr_accessor :name_id_formats_opts
+    attr_accessor :asserted_attributes_opts
 
-    def initialize(reference_id,
-          response_id,
-          issuer_uri,
-          principal,
-          audience_uri,
-          saml_request_id,
-          saml_acs_url,
-          algorithm,
-          authn_context_classref,
-          expiry=60*60,
-          encryption_opts=nil,
-          session_expiry=0,
-          signed_message_opts
-          )
+    def initialize(
+        reference_id,
+        response_id,
+        issuer_uri,
+        principal,
+        audience_uri,
+        saml_request_id,
+        saml_acs_url,
+        algorithm,
+        authn_context_classref,
+        expiry=60*60,
+        encryption_opts=nil,
+        session_expiry=0,
+        signed_message_opts,
```

Comment:
This is causing issues on my local - perhaps we should move the nil-defaults above this? without deeper investigation, I don't know what the effects of putting a default on `signed-message-opts` would do

Reply:
Can you share your local issue here if you still have your environment in your local.
I have got your concern about default value. But for now we are setting default values in high levels of the module.
`signed_message_opts = opts[:signed_message] || false` - lib/saml_idp/controller.rb
And it's not really related to this PR, it might be good to have different "issue" and work as an improvement.

Reply:
specs don't run at all:

```
An error occurred while loading ./spec/lib/saml_idp/signed_info_builder_spec.rb.
Failure/Error: require 'saml_idp/saml_response'

SyntaxError:
  /Users/jon.phenow/Code/saml_idp/lib/saml_idp/saml_response.rb:38: syntax error, unexpected '=', expecting ')'
          name_id_formats_opts = nil,
                               ^
  /Users/jon.phenow/Code/saml_idp/lib/saml_idp/saml_response.rb:38: Can't assign to nil
  ...     name_id_formats_opts = nil,
  ...                            ^~~
  /Users/jon.phenow/Code/saml_idp/lib/saml_idp/saml_response.rb:40: syntax error, unexpected ')', expecting `end'
      )
      ^
  /Users/jon.phenow/Code/saml_idp/lib/saml_idp/saml_response.rb:104: syntax error, unexpected `end', expecting end-of-input
# ./lib/saml_idp.rb:4:in `<module:SamlIdp>'
# ./lib/saml_idp.rb:2:in `<top (required)>'
# ./spec/spec_helper.rb:18:in `<top (required)>'
# ./spec/lib/saml_idp/signed_info_builder_spec.rb:1:in `<top (required)>'```

Can't have a series of variables with defaults and one in the middle of them that doesn't have a default.

Reply:
Shame on me. I'm sorry it was a pretty obvious one.
![shame_meme](https://user-images.githubusercontent.com/19234717/120915940-98e4ba00-c6e1-11eb-8308-706b7764d16b.jpeg)